parsers/bibtex.c

```diff
@@ -0,0 +1,459 @@
+	TOKEN_OPEN_CURLY = '{',
+	TOKEN_CLOSE_CURLY = '}',
+	/* above is special types */
+	TOKEN_UNDEFINED = 256,
+	TOKEN_KEYWORD,
+	TOKEN_IDENTIFIER,
+	TOKEN_STRING,
+};
+typedef int tokenType;
+
+typedef struct sTokenInfo {
+	tokenType		type;
+	keywordId		keyword;
+	vString *		string;
+	vString *		scope;
+	unsigned long 	lineNumber;
+	MIOPos 			filePosition;
+} tokenInfo;
+
+/*
+ *	DATA DEFINITIONS
+ */
+
+static langType Lang_bib;
+
+typedef enum {
+	BIBTAG_ARTICLE,
+  BIBTAG_BOOK,
+  BIBTAG_BOOKLET,
+  BIBTAG_CONFERENCE,
+  BIBTAG_INBOOK,
+  BIBTAG_INCOLLECTION,
+  BIBTAG_INPROCEEDINGS,
+  BIBTAG_MANUAL,
+  BIBTAG_MASTERSTHESIS,
+  BIBTAG_MISC,
+  BIBTAG_PHDTHESIS,
+  BIBTAG_PROCEEDINGS,
+  BIBTAG_TECHREPORT,
+  BIBTAG_UNPUBLISHED,
+	BIBTAG_COUNT
+} bibKind;
+
+static kindDefinition BibKinds [] = {
+	{ true,  'a', "article",			  "article"		   },
+	{ true,  'b', "book",			      "book"		     },
+	{ true,  'B', "booklet",			  "booklet"		   },
+	{ true,  'c', "conference",		  "conference"	 },
+	{ true,  'i', "inbook",			    "inbook"		   },
+	{ true,  'I', "incollection",	  "incollection" },
+	{ true,  'j', "inproceedings",	"inproceedings"},
+	{ true,  'm', "manual",	  		  "manual"		   },
+	{ true,  'M', "mastersthesis",  "mastersthesis"},
+	{ true,  'n', "misc",   			  "misc"  		   },
+	{ true,  'p', "phdthesis",			"phdthesis"	   },
+	{ true,  'P', "proceedings",		"proceedings"  },
+	{ true,  't', "techreport",		  "techreport"   },
+	{ true,  'u', "unpublished",	  "unpublished"  }
+};
+
+static const keywordTable BibKeywordTable [] = {
+	/* keyword			  keyword ID */
+	{ "article",	    KEYWORD_article				},
+	{ "book",	        KEYWORD_book				  },
+	{ "booklet",	    KEYWORD_booklet				},
+	{ "conference",	  KEYWORD_conference		},
+	{ "inbook",	      KEYWORD_inbook				},
+	{ "incollection",	KEYWORD_incollection	},
+	{ "inproceedings",KEYWORD_inproceedings	},
+	{ "manual",	      KEYWORD_manual				},
+	{ "mastersthesis",KEYWORD_mastersthesis	},
+	{ "misc",	        KEYWORD_misc				  },
+	{ "phdthesis",	  KEYWORD_phdthesis			},
+	{ "proceedings",	KEYWORD_proceedings		},
+	{ "techreport",	  KEYWORD_techreport		},
+	{ "unpublished",	KEYWORD_unpublished		}
+};
+  
+/*
+ *	 FUNCTION DEFINITIONS
+ */
+
+static tokenInfo *newToken (void)
+{
+	tokenInfo *const token = xMalloc (1, tokenInfo);
+
+	token->type			= TOKEN_UNDEFINED;
+	token->keyword		= KEYWORD_NONE;
+	token->string		= vStringNew ();
+	token->scope		= vStringNew ();
+	token->lineNumber   = getInputLineNumber ();
+	token->filePosition = getInputFilePosition ();
+
+	return token;
+}
+
+static void deleteToken (tokenInfo *const token)
+{
+	vStringDelete (token->string);
+	vStringDelete (token->scope);
+	eFree (token);
+}
+
+/*
+ *	 Tag generation functions
+ */
+static void makeBibTag (tokenInfo *const token, bibKind kind)
+{
+	if (BibKinds [kind].enabled)
+	{
+		const char *const name = vStringValue (token->string);
+		int parentKind = KIND_GHOST_INDEX;
+		vString *parentName = vStringNew();
```

Comment:
parentName is not used. parentKnd is not uesd. Both can be remoed.

Reply:
these were removed in 952f651